### ISSUE TITLE
Deprecation of HTTP Basic Auth and OAuth 1.0a

### ISSUE THREAD

**Issue by opk12** [viking-gps/viking#196]
Thank you for this program. Does it use OAuth 1? OSM is slowly deprecating HTTP Basic Auth and OAuth 1.0a in favor of OAuth 2 (see [announcement](https://www.openstreetmap.org/user/pnorman/diary/401157) and [tracking issue](https://github.com/openstreetmap/operations/issues/867)) so I thought you'd want to know.

**Comment by rnorris** [comment]
Yes I'm aware and have been thinking about this for a while - last time I looked (many years ago) - there wasn't many/any C library options seemingly available. But now there are:

https://github.com/babelouest/iddawc

or

https://github.com/OpenIDC/liboauth2

Still obviously need to get around to doing it...

**Comment by danieldegroot2** [comment]
Note, `api.openstreetmap.org` API url currently can not do OAuth 2 handshake. See https://github.com/openstreetmap/operations/issues/951#issuecomment-1748717154
You can split the `www.` base URL and `api.` API url. See https://github.com/osmlab/osm-auth/pull/123#issuecomment-1777330094

cc: @rnorris

**Comment by CrabstickOMalley** [comment]
It seems at this point there's no way for an OSM user to create Oauth 1.0 tokens anymore.   So there is no way for a 'new' user to use Viking with OSM since they migrated to Oauth 2, although from what I gather they have not 'cut off' legacy apps that have valid tokens already so I guess those with working installs are OK temporarily.   I was a user of Viking years ago and wanted to re-start but at this point having no tokens means apparently OSM integration is impossible.   Unfortunate but seems like Viking as an application is in 'sunset' mode anyway.  If anyone can recommend a good replacement with live ongoing maintenance/development that would be great. Thanks.

**Comment by rnorris** [comment]
For the moment use of specific OSM account operations: Acquire / Upload (My) OSM Traces are disabled.
The issue is left open for (if/when) an OAuth2 solution is implemented.

**Comment by guyou** [comment]
Note: here is an example https://gitlab.gnome.org/GNOME/librest/-/blob/master/examples/gitlab-oauth2-example.c?ref_type=heads

**Comment by guyou** [comment]
I started some experiment and having an initial understanding of the behavior: https://github.com/guyou/viking/blob/experiment-osm-oauth2/src/osm-oauth2-example.c

The full example to follow : Gnome Maps (see https://gitlab.gnome.org/GNOME/gnome-maps/-/blob/main/src/osmConnection.js?ref_type=heads)
The main idea (you can experiment if you have Gnome Maps installed):
1. you request an authentification
2. an web browser open, expecting validation
3. you accept and a new page display a token
4. you copy this token into the app (here Gnome Maps, but would also be Viking)
5. Your app is now able to authenticate agains OSM servers

**Comment by guyou** [comment]
Update: the Oauth2 feature is available in librest >= 0.9. Thus, only Ubuntu 24.04 proposes this version.
